play/src/front/Rules/PersonalAreaPropertyEditorRules/PersonalAreaPropertyEditorRules.ts

```diff
@@ -0,0 +1,102 @@
+import { PersonalAreaAccessClaimMode, PersonalAreaPropertyData } from "@workadventure/map-editor";
+import { derived, get, Readable } from "svelte/store";
+import { mapEditorSelectedAreaPreviewStore } from "../../Stores/MapEditorStore";
+
+export type Option = {
+    value: string;
+    label: string;
+    created: undefined | boolean;
+};
+
+export class PersonalAreaPropertyEditorRules {
+    private readonly selectedAreaPersonalAreaPropertyStore: Readable<PersonalAreaPropertyData | undefined>;
+    private readonly defaultPersonalAreaProperty: PersonalAreaPropertyData = {
+        id: crypto.randomUUID(),
+        type: "personalAreaPropertyData",
+        accessClaimMode: "dynamic",
+        allowedTags: [],
+        ownerId: "",
+    };
+
+    constructor() {
+        this.selectedAreaPersonalAreaPropertyStore = derived(
+            mapEditorSelectedAreaPreviewStore,
+            (currentAreaPreview) => {
+                if (currentAreaPreview === undefined) {
+                    return;
+                }
+                for (const areaPropertyData of currentAreaPreview.getProperties()) {
+                    if (areaPropertyData.type === "personalAreaPropertyData") {
+                        return areaPropertyData;
+                    }
+                }
+                currentAreaPreview.addProperty({ ...this.defaultPersonalAreaProperty });
+                return this.defaultPersonalAreaProperty;
+            }
+        );
+    }
+
+    get defaultAllowedTags(): Option[] {
+        return [
+            {
+                value: "member",
+                label: "member",
+                created: false,
+            },
+            {
+                value: "admin",
+                label: "admin",
+                created: false,
+            },
+        ];
+    }
+
+    public mapExistingAllowedTagsToTagOption(allowedTags: string[]): Option[] {
+        return allowedTags.map((allowedTag) => ({
+            value: allowedTag,
+            label: allowedTag,
+            created: false,
+        }));
+    }
+
+    public getPersonalAreaPropertyDataUnSubscriber(
```

Comment:
Weird name. You don't get a "unsubscriber". This is merely a side-effect of subscribing to the selectedAreaPersonalAreaPropertyStore.